frontend/src/metabase/visualizations/components/settings/ChartSettingInputSuggestion.tsx

```diff
@@ -0,0 +1,62 @@
+import React, { useState } from "react";
+import AutocompleteInput from "metabase/core/components/AutocompleteInput";
+
+interface ChartSettingInputSuggestionProps {
+  value: string;
+  onChange: (value: string) => void;
+  id?: string;
+  options?: string[];
+}
+
+const matchPattern = /.*{{([^{}]*)$/;
+
+const filterFn = (value: string | undefined, options: string[]) => {
+  if (options && value) {
+    const match = value.match(matchPattern);
+    if (match) {
+      const suggestionFilter = match[1];
+      return options.filter(option =>
+        option.toLowerCase().includes(suggestionFilter.toLowerCase()),
+      );
+    }
+  }
+  return [];
+};
+
+const ChartSettingInputSuggestion = ({
```

Comment:
Since this component only works for links, should it be renamed accordingly? E.g. `ChartSettingLinkUrlInput`.

Reply:
I believe we also need to add a few unit tests here, at least for the happy path.

Reply:
And I'd add a e2e test (or modify an existing one if we have it) to make it select the column from the autocomplete list and make sure the result link actually works (e.g. the variable is substituted correctly).